File: test/io/metrics.rb
# frozen_string_literal: true

# Released under the MIT License.
# Copyright, 2026, by Samuel Williams.

require "io/metrics"

describe IO::Metrics do
	it "has a version number" do
		expect(IO::Metrics::VERSION).to be =~ /\A\d+\.\d+\.\d+\Z/
	end
end

File: ext/io/metrics/metrics.c
// Released under the MIT License.
// Copyright, 2026, by Samuel Williams.

#include <ruby.h>

#if defined(HAVE_LINUX_INET_DIAG_H) || defined(__linux__)
void Init_IO_Metrics_Listener(VALUE IO_Metrics);
#endif

void Init_IO_Metrics(void)
{
	VALUE IO_Metrics = rb_define_module_under(rb_cIO, "Metrics");
	
#if defined(HAVE_LINUX_INET_DIAG_H) || defined(__linux__)
	Init_IO_Metrics_Listener(IO_Metrics);
#endif
}
